scripts/provisioning/s3_haproxy_config.py

```diff
@@ -117,7 +117,166 @@ def process(self):
+    h1-case-adjust accept-encoding Accept-Encoding
+    h1-case-adjust action Action
+    h1-case-adjust credential Credential
+    h1-case-adjust signedHeaders SignedHeaders
+    h1-case-adjust signature Signature
+    h1-case-adjust clientAbsoluteUri ClientAbsoluteUri
+    h1-case-adjust clientQueryParams ClientQueryParams
+    h1-case-adjust connection Connection
+    h1-case-adjust host Host
+    h1-case-adjust method Method
+    h1-case-adjust request_id Request_id
+    h1-case-adjust requestorAccountId RequestorAccountId
+    h1-case-adjust requestorAccountName RequestorAccountName
+    h1-case-adjust requestorCanonicalId RequestorCanonicalId
+    h1-case-adjust requestorEmail RequestorEmail
+    h1-case-adjust requestorUserId RequestorUserId
+    h1-case-adjust requestorUserName RequestorUserName
+    h1-case-adjust s3Action S3Action
+    h1-case-adjust user-Agent User-Agent
+    h1-case-adjust version Version
+    h1-case-adjust x-Amz-Content-SHA256 X-Amz-Content-SHA256
+    h1-case-adjust x-Amz-Date X-Amz-Date
+    h1-case-adjust x-Forwarded-For X-Forwarded-For
+
+    chroot      /var/lib/haproxy
+    pidfile     /var/run/haproxy.pid
+    maxconn     13
+    user        haproxy
+    group       haproxy
+    nbproc      2
+    daemon
+
+    # turn on stats unix socket and dynamic configuration reload
+    stats socket /var/lib/haproxy/stats expose-fd listeners
+
+    # utilize system-wide crypto-policies
+    # ssl-default-bind-ciphers PROFILE=SYSTEM
+    # ssl-default-server-ciphers PROFILE=SYSTEM
+
+    #SSL options            # utilize system-wide crypto-policies
+    tune.ssl.default-dh-param 2048
+'''
+    default_text = '''
+#---------------------------------------------------------------------
+# common defaults that all the 'listen' and 'backend' sections will
+# use if not designated in their block
+#---------------------------------------------------------------------
+defaults
+    mode                    http
+    log                     global
+    option                  httplog
+    option                  dontlognull
+    option h1-case-adjust-bogus-server
+    option h1-case-adjust-bogus-client
+    option http-server-close
+    option forwardfor       except 127.0.0.0/8
+    option                  redispatch
+    errorfile               503 /etc/haproxy/errors/503.http
+    retries                 3
+    timeout http-request    10s
+    timeout queue           10s
+    timeout connect         5s
+    timeout client          360s
+    timeout server          360s
+    timeout http-keep-alive 10s
+    timeout check           10s
+    maxconn                 3000
+'''
+    frontend_s3main_text = '''
+#----------------------------------------------------------------------
+# FrontEnd S3 Configuration
+#----------------------------------------------------------------------
+frontend s3-main
+    # s3 server port
+    bind 0.0.0.0:80
+    bind 0.0.0.0:443 ssl crt /etc/cortx/s3/stx/stx.pem
```

Comment:
1)Make sure that right version of self signed certificate is packaged in RPM and is installed by default in /etc/ssl (INSTALL_LOC).
2)Add code to copy from INSTALL_LOC to TGT_LOC

Reply:
TGT_LOC = CONFSTORE_S3_SECURITY_CERTIFICATE: "cortx>common>security>ssl_certificate"